package.json

```diff
@@ -92,23 +92,24 @@
     "!__tests__"
   ],
   "scripts": {
-    "docs": "npm run build && npm run docs:ci",
-    "docs:ci": "npm run docs:extract -- --local && npm run docs:generate",
-    "docs:generate": "api-documenter markdown --input-folder ./tmp --output-folder ./api/docs",
-    "docs:extract": "api-extractor run --verbose",
-    "build": "npm run clean && tsc",
-    "clean": "rm -rf dist",
-    "version:version-file": "genversion --template version-template.ejs --force lib/version.ts",
-    "version": "npm run version:version-file && npm run docs && git add ./lib/version.ts ./api/docs",
-    "test": "npm run test:ts && npm run test:legacy",
-    "test:ts": "jest",
-    "test:legacy": "npm run build -- --declaration false && ./test/scripts/vows.js test/*_test.js",
-    "typecheck": "tsc --noEmit",
-    "cover": "jest --coverage",
-    "lint": "eslint .",
-    "eslint": "eslint .",
-    "prettier": "prettier .",
-    "format": "npm run eslint -- --fix"
+    "build": "npm run _build:clean && npm run _build:compile",
+    "precommit": "npm run build && npm run _api:update && npm run _docs:generate && npm run _format:fix && npm run _lint:fix",
+    "prepack": "npm run build",
+    "test": "npm run build && npm run _test:ts && npm run _test:legacy",
+    "version": "npm run _version:generate && npm run precommit && git add --renormalize .",
+    "_api:check": "api-extractor run --verbose",
+    "_api:update": "api-extractor run --verbose --local",
+    "_build:clean": "rm -rf dist",
+    "_build:compile": "tsc",
+    "_docs:generate": "api-documenter markdown --input-folder ./tmp --output-folder ./api/docs",
+    "_docs:fix": "prettier ./api/docs --write",
+    "_format:check": "prettier . --check",
+    "_format:fix": "prettier . --write",
+    "_lint:check": "eslint .",
```

Comment:
I also debated these. Most IDE integration can handle linting & formatting tasks automatically for you which reduces the need to call these scripts as part of your regular workflow. This is also why I opted for exposing the single `precommit` scripts which handles linting, formatting, and rebuilding docs so there are fewer regular scripts to remember. 

I thought this might be a point of contention though. So you'd still prefer individual scripts like `lint` and `format` rather than putting them under a single `precommit` (or `prepare-pr`) script instead?

Reply:
I don't care about `format`, but sometimes I still run `npm run lint` manually.

Reply:
👍 Added `lint` back